### DISCUSSION TITLE
Reusing Fragments

### DISCUSSION THREAD

**Discussion by wigandt-dot** [CMPUT301F21T22/Nabu#16] [Ideas]
I noticed that some fragments, more specifically the ones relating to confirming an action, were quite similar:
- https://github.com/CMPUT301F21T22/Nabu/wiki/Mockup:-Habits-(Habit-Delete-Confirmation)
- https://github.com/CMPUT301F21T22/Nabu/wiki/Mockup:-Habits-(Habit-Event-Delete-Confirmation)
- https://github.com/CMPUT301F21T22/Nabu/wiki/Mockup:-Settings-(Reset-Confirmation)
- https://github.com/CMPUT301F21T22/Nabu/wiki/Mockup:-Settings-(Logout-Confirmation)

Might it be feasible to combine at least some of these into a single reusable fragment, so that we just change the warning text between them? Or would it be much easier or more organized to put them as separate fragments?

**Comment by trnglina** [comment]
Yes, we can have a simple dialog class and then pass in the individual messages, and affirmative/cancel button strings.
